_posts/2022-11-01-setup-a-ci-pipeline-using-github-actions-and-docker.md

```diff
@@ -0,0 +1,188 @@
+
+Select the GCC Makefile option on Change Project Generators, save, and Force Generation.
+
+![Makefile Generator](/img/setup-a-ci-pipeline-using-github-actions-and-docker/7-gcc-makefile-gen.png)
+
+After that, the empty.Makefile and empty.project.mak files will be created in the project root directory.
+
+![Makefiles](/img/setup-a-ci-pipeline-using-github-actions-and-docker/8-makefile.png)
+
+Now you can build the project in the terminal since Simplicity Studio installed the toolchain.
+
+If you are running it on Linux, you can go directly to the terminal, but if you are using Windows, you will need to install [Cygwin](https://www.cygwin.com/) and execute the command.
+
+```bash
+make -f empty.Makefile
+```
+
+You, at this point, can compile the project at your local machine using the terminal.
+
+![Makefiles](/img/setup-a-ci-pipeline-using-github-actions-and-docker/9-compile.png)
+
+# Test the compilation on a Docker Container
+
+We will use a docker image with Simplicity Studio to build our project on a container. This will allow us to configure GitHub Actions to use the docker image to do the same.
+
+So let's pull the image to our computer:
+
+```powershell
+docker pull leoribg/simplicity-studio-5:latest
+```
+
+Start the container from the image:
+
+```powershell
+docker run -t -d leoribg/simplicity-studio-5
+```
+
+Copy the project files to the container:
+
+```powershell
+docker cp C:\Users\leonardo\SimplicityStudio\v5_workspace\empty 1027ffa9c954:/project
+```
+
+Compile the project and make a static code analysis:
+
+```bash
+cd /project/
+make -f empty.Makefile -j8
+cppcheck . --output-file=cpplog.txt
+```
+
+## Create the Git Repository
+
+Since the project was created, let's create a git repository. 
+
+For this tutorial, this is the repository I've used for my project:
+
+ ‣
+
+To start and add your project to a git repository, run the following commands on your project directory:
+
+```bash
+git init
+git add .
+git commit -m "first commit"
+git remote add origin git@github.com:"user"/"repository".git
+git push -u origin master
+```
+
+# Configure the GitHub Actions
+
+You can start to configure the GitHub Actions for your project by creating the .yml file with your pipeline inside the “.github/workflows” folder. Or, you can create at your repository page by creating the file following a template provided by GitHub. Let's do the second option, that's easier for those who are starting.
+
+So, go to your repository page and click on the “Actions” tab:
+
+![GitHub Actions](/img/setup-a-ci-pipeline-using-github-actions-and-docker/10-actions.png)
+
+Choose the right template:
+
+![Actions Template](/img/setup-a-ci-pipeline-using-github-actions-and-docker/11-actions-template.png)
+
+This tutorial will use a single workflow to execute our jobs. So, it will only have a single .yml file.
+
+The .yml file starts like this, we will modify it.
+
+![Cpp Workflow](/img/setup-a-ci-pipeline-using-github-actions-and-docker/12-cpp-workflow.png)
```

Comment:
Let's use the actual contents of this file instead of an image- it will make the post much more readable (especially on mobile)

Reply:
+1

Reply:
Ok!